### PR TITLE
Add support to quick_lint_js

### PR DESCRIPTION
**Pull request by tiagovla**
https://quick-lint-js.com/

### CODE REVIEW

File: lua/lspconfig/quick_lint_js.lua [hunk 1]
```diff
@@ -0,0 +1,32 @@
+local configs = require("lspconfig/configs")
+local util = require("lspconfig/util")
+
+configs.quick_lint_js = {
+  default_config = {
+    cmd = {"quick-lint-js", "--lsp-server"},
+    filetypes = {"javascript"},
```
Comment (lithammer) [inline lua/lspconfig/quick_lint_js.lua:7]: Add `javascriptreact` which is a common filetype for `.jsx` files. Does it work for TypeScript projects as well?

```suggestion
    filetypes = {"javascript", "javascriptreact"},
```

File: lua/lspconfig/quick_lint_js.lua [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    root_dir = function(fname)
+      return util.find_node_modules_ancestor(fname) or
+        util.find_git_ancestor(fname) or util.path.dirname(fname)
```
Comment (lithammer) [inline lua/lspconfig/quick_lint_js.lua:10]:
```suggestion
        util.find_git_ancestor(fname) or
        util.path.dirname(fname)
```

File: lua/lspconfig/quick_lint_js.lua [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        util.find_node_modules_ancestor(fname) or
+        util.find_git_ancestor(fname) or
+        util.path.dirname(fname)
+    end
+  },
+  docs = {
+    description = [[
+https://quick-lint-js.com/
+
+quick-lint-js finds bugs in JavaScript programs.
```
Comment (mjlbach) [inline lua/lspconfig/quick_lint_js.lua:19]: This sounds way too much like an advertisement.

Comment (tiagovla) [reply lua/lspconfig/quick_lint_js.lua:19]: @mjlbach fixed it.

**Comment by mjlbach** [comment]
Can you rebase, and check that this passes style and commit lint? Thanks!

**Pull request closed without merging**